crates/services/executor/src/executor.rs

```diff
@@ -648,18 +667,115 @@ where
                             .events
                             .push(ExecutorEvent::MessageImported(message));
                     }
-                    Event::Transaction(_) => {
-                        // TODO: implement handling of forced transactions in a later PR
+                    Event::Transaction(relayed_tx) => {
+                        let id = relayed_tx.id();
+                        // perform basic checks
+                        let checked_tx_res = Self::validate_forced_tx(
+                            relayed_tx,
+                            header,
+                            &self.consensus_params,
+                        );
+                        // handle the result
+                        match checked_tx_res {
+                            Ok(checked_tx) => {
+                                checked_forced_txs.push(checked_tx);
+                            }
+                            Err(err) => {
+                                execution_data.events.push(
+                                    ExecutorEvent::ForcedTransactionFailed {
+                                        id,
+                                        block_height,
+                                        failure: err.to_string(),
+                                    },
+                                );
+                            }
+                        }
                     }
                 }
             }
         }
 
         execution_data.event_inbox_root = root_calculator.root().into();
 
+        Ok(checked_forced_txs)
+    }
+
+    /// Parse forced transaction payloads and perform basic checks
+    fn validate_forced_tx(
+        relayed_tx: RelayedTransaction,
+        header: &PartialBlockHeader,
+        consensus_params: &ConsensusParameters,
+    ) -> Result<CheckedTransaction, ForcedTransactionFailure> {
+        let parsed_tx = Self::parse_tx_bytes(&relayed_tx)?;
+        let checked_tx =
+            Self::get_checked_tx(parsed_tx, *header.height(), consensus_params)?;
+        Self::tx_is_valid_variant(&checked_tx)?;
+        Self::relayed_tx_claimed_enough_max_gas(&checked_tx, &relayed_tx)?;
```

Comment:
We need to do this check even before `Self::get_checked_tx`, because `into_checked` may be heavy in the case of `Create` transactions.